src/status_im/wallet/accounts/core.cljs

```diff
@@ -13,78 +13,222 @@
             [status-im.utils.types :as types]
-            [status-im.wallet.core :as wallet]))
+            [status-im.wallet.core :as wallet]
+            [clojure.string :as string]
+            [status-im.utils.security :as security]
+            [status-im.multiaccounts.recover.core :as recover]))
+
+(fx/defn start-adding-new-account
+  {:events [:wallet.accounts/start-adding-new-account]}
+  [{:keys [db] :as cofx} {:keys [type] :as add-account}]
+  (let [{:keys [latest-derived-path]} (:multiaccount db)
+        path-num (inc latest-derived-path)
+        account  (merge
+                  {:color (rand-nth colors/account-colors)}
+                  (when (= type :generate)
+                    {:name (str "Account " path-num)}))]
+    (fx/merge cofx
+              {:db (assoc db :add-account (assoc add-account :account account))}
+              (navigation/navigate-to-cofx :add-new-account nil))))
+
+(fx/defn new-account-error
+  {:events [::new-account-error]}
+  [{:keys [db]} error-key error]
+  {:db (update db :add-account merge {error-key error
+                                      :step nil})})
+
+(defn account-stored [path type]
+  (fn [result]
+    (let [{:keys [error publicKey address]} (types/json->clj result)]
+      (if error
+        (re-frame/dispatch [::new-account-error :account-error error])
+        (re-frame/dispatch [:wallet.accounts/account-stored
+                            {:address    address
+                             :public-key publicKey
+                             :type       type
+                             :path       path}])))))
+
+(def dec-pass-error "could not decrypt key with given password")
+
+(defn derive-and-store-account [path hashed-password type]
+  (fn [value]
+    (let [{:keys [id error]} (types/json->clj value)]
+      (if error
+        (re-frame/dispatch [::new-account-error :password-error error])
+        (status/multiaccount-derive-addresses
+         id
+         [path]
+         (fn [_]
+           (status/multiaccount-store-derived
+            id
+            [path]
+            hashed-password
+            (fn [result]
+              (let [{:keys [error] :as result} (types/json->clj result)
+                    {:keys [publicKey address]} (get result (keyword path))]
+                (if error
+                  (re-frame/dispatch [::new-account-error :account-error error])
+                  (re-frame/dispatch
+                   [:wallet.accounts/account-stored
+                    {:address    address
+                     :public-key publicKey
+                     :type       type
+                     :path       (if (string/starts-with? path "m/")
+                                   (str constants/path-wallet-root
+                                        "/" (last (string/split path "/")))
+                                   path)}])))))))))))
+
+(def pass-error "cannot retrieve a valid key for a given account: could not decrypt key with given password")
```

Comment:
Is that returned form status-go side? We probably need to start using error codes or something, because comparing messages might not be the best idea and definitely isn't a very reliable way.